R/plotCCA.R

```diff
@@ -0,0 +1,504 @@
+#' 
+#' @param arrow_size Number specifying the size of arrows (Defaults: 0.25).
+#' 
+#' @param text_size Number specifying the size of text and labels (Default: 4).
+#' 
+#' @param text_colour String specifying the colour of text and labels (Default: "black").
+#' 
+#' @param text_color Alias for `text_colour`.
+#' 
+#' @param vec_text If TRUE, text instead of labels are used to label vectors (Default: TRUE).
+#' 
+#' @param repel_text If TRUE, labels are repelled (Default: TRUE).
+#' 
+#' @param ... additional parameters for plotting, inherited from
+#'   \code{\link[scater:plotReducedDim]{plotReducedDim}},
+#'   \code{\link[ggplot2:geom_label]{geom_label}} and
+#'   \code{\link[ggrepel:geom_label_repel]{geom_label_repel}}.
+#' 
+#' @details
+#' plotRDA and plotCCA create an RDA/CCA plot starting from the output of
+#' \code{\link[mia:runCCA]{CCA and RDA}} functions, two common methods for
+#' supervised ordination of microbiome data. Both
+#' \code{\link[SummarizedExperiment:SummarizedExperiment-class]{SummarizedExperiment}}
+#' and rda/cca objects are supported as input. When the input is a SummarizedExperiment
+#' or a TreeSummarizedExperiment, this should contain the output of runRDA
+#' in the reducedDim slot and the argument \code{dimred} needs to be defined.
+#' When the input is an rda/cca object, this should be returned from
+#' calculateRDA and the argument \code{dimred} needs not be defined. However,
+#' the first method is recommended because it provides the option to adjust
+#' aesthetics to the colData variables through the arguments inherited from
+#' \code{\link[scater:plotReducedDim]{plotReducedDim}}.
+#' 
+#' @return 
+#' A \code{ggplot2} object 
+#'
+#' @name plotRDA
+#'
+#' @examples
+#' 
+#'  # Load dataset
+#'  library(miaViz)
+#'  data(enterotype)
+#'  tse <- enterotype
+#'  
+#'  # Perform RDA
+#'  tse <- runRDA(tse,
+#`                formula = assay ~ ClinicalStatus + Gender + Age,
+#'                FUN = vegan::vegdist,
+#'                distance = "bray")
+#'                
+#'  # Create RDA plot coloured by variable
+#'  plotRDA(tse, "RDA",
+#'          colour_by = "ClinicalStatus")
+#'  
+#'  # Create RDA plot with empty ellipses
+#'  plotRDA(tse, "RDA",
+#'          colour_by = "ClinicalStatus,
+#'          ellipse_fill = FALSE)
+#'  
+#'  # Create RDA plot with text encased in labels
+#'  plotRDA(tse, "RDA",
+#'          colour_by = "ClinicalStatus,
+#'          vec_text = FALSE)
+#'  
+#'  # Create RDA plot without repelling text
+#'  plotRDA(tse, "RDA",
+#'          colour_by = "ClinicalStatus,
+#'          repel_text = FALSE)
+#'  
+#'  
+NULL
+
+#' @rdname plotRDA
+#' @aliases plotRDA
+#' @export
+setGeneric("plotCCA", signature = c("object"),
+           function(object, dimred, ...) standardGeneric("plotCCA"))
+
+#' @rdname plotRDA
+#' @aliases plotRDA
+#' @export
+setMethod("plotCCA", signature = c(object = "SingleCellExperiment"),
+    function(object, dimred, ...){
```

Comment:
This could be just `function(object, ...){`

Reply:
This throws an error if changed. Isn't `dimred` also a compulsory arg?

Reply:
I was thinking if plotCCA does not have any parameters than `object` --> so there is as little as possible and everything happens in plotRDA

Reply:
We can consider moving dimred parameter to `.incorporate_rda_vis`

However, because it is always needed, it might be easier to be in plotRDA since then user can just type plotRDA(tse, "RDA") and not plotRDA(tse, dimred = "RDA")

Reply:
Yes however if `dimred` is not included also in the args of `plotCCA`, it throws an error:

```

plotCCA(tse, "RDA")
# Error in plotRDA(object, ...) :
# argument "dimred" is missing, with no default
```

Reply:
Whatever the solution will be we should make sure that it is systematic across the family of similar functions.

I do not see great immediate need for supporting dimred here but there can be use cases (e.g. using dimension reduction as a rough denoising for the data).

Reply:
--> yes, we can keep it here in this case